File: fluttermultiplatform/lib/models/shipment_log.dart
class ShipmentLog {
  final String id;
  final String shipmentId;
  final String status;
  final String location;
  final String notes;
  final String timestamp;

  ShipmentLog({required this.id, this.shipmentId = '', this.status = '', this.location = '', this.notes = '', this.timestamp = ''});

  factory ShipmentLog.fromJson(Map<String, dynamic> json) => ShipmentLog(
        id: json['id']?.toString() ?? '',
        shipmentId: json['shipment_id']?.toString() ?? '',
        status: json['status'] ?? '',
        location: json['location'] ?? '',
        notes: json['notes'] ?? '',
        timestamp: json['timestamp'] ?? '',
      );
}

File: backend-spring/src/main/java/com/cronos/entity/ShipmentLog.java
package com.cronos.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLRestriction;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Table(name = "shipment_logs")
@SQLDelete(sql = "UPDATE shipment_logs SET deleted_at = NOW() WHERE id = ?")
@SQLRestriction("deleted_at IS NULL")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class ShipmentLog {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "uuid")
    private UUID id;

    @JsonProperty("shipment_id")
    @Column(name = "shipment_id", nullable = false, columnDefinition = "uuid")
    private UUID shipmentId;

    @Column(nullable = false, length = 50)
    private String status;

    @Column(length = 255)
    private String location;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(nullable = false)
    private LocalDateTime timestamp;

    @JsonProperty("created_at")
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @JsonIgnore
    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "shipment_id", insertable = false, updatable = false)
    @JsonIgnore
    private Shipment shipment;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

